### PR TITLE
Alerts

### PR DESCRIPTION
**Pull request by mxsnyder**
As usual, ignore everything in the "dist" folder.

### CODE REVIEW

File: js/alert-box.js [hunk 1]
```diff
@@ -14,28 +14,69 @@ bcpl.alertBox = (($) => {
...
+					(data) => data ? callback(data, true) : callback(undefined, false),
+					() => callback(undefined, false)
+				);
+		} else {
+			console.error('A missing or invalid callback has been supplied.');
+		}
+	};
+
+	const hideNotificationBar = ($container) => {
+		$container.addClass('dismissed');
```
Comment (martypowell) [inline js/alert-box.js]: chain.

File: js/alert-box.js [hunk 2]
```diff
@@ -14,28 +14,69 @@ bcpl.alertBox = (($) => {
...
+		}
+
+		displayNotificationBar(!alertData);
+	};
+
+	const getAlertDescription = (callback) => {
+		if (callback && typeof callback === 'function') {
+			$.ajax(CONSTANTS.baseApiUrl + CONSTANTS.shared.urls.alertNotification)
+				.then(
+					(data) => data ? callback(data, true) : callback(undefined, false),
```
Comment (martypowell) [inline js/alert-box.js]: would consider making these methods and then passing them in (this is my preference but not a deal breaker)
```javascript
const onSuccess = () => { /* stuff goes here */ };
const onFailture  = () => { /* stuff goes here */ };
...
.then(onSuccess, onFailure);
```

File: js/alert-box.js [hunk 3]
```diff
@@ -14,28 +14,69 @@ bcpl.alertBox = (($) => {
...
+			console.error('A missing or invalid callback has been supplied.');
+		}
+	};
+
+	const hideNotificationBar = ($container) => {
+		$container.addClass('dismissed');
+		$container.show();
+	};
+
+	const displayNotificationBar = (shouldHide) => {
```
Comment (martypowell) [inline js/alert-box.js]: It looks like ```hideNotificationBar``` could be called twice (lines 57 and 65)? Is that desired?

Comment (mxsnyder) [reply js/alert-box.js]: Great catch! fixed.

File: js/alert-box.js (hunk 1 above)
Comment (mxsnyder) [reply js/alert-box.js]: Done

File: js/alert-box.js (hunk 2 above)
Comment (mxsnyder) [reply js/alert-box.js]: Done, though since I'm passing through a callback, I still had to use function bodies.

**Comment by mxsnyder** [comment]
Ready for a second look. Thanks!

**Pull request merged**